ui/src/pages/QuickSetup/AccountSelectionModal.tsx

```diff
@@ -0,0 +1,262 @@
+              ? accounts.find(a => a.id.label === accountNamesState[k])
+              : allocationAccounts.find(a => a.id.label === accountNamesState[k]);
+          return account;
+        });
+        onFinish(accts);
+        setAccountNamesState(emptyNamesState);
+        setOpen(false);
+      }}
+    >
+      <Modal.Header as="h2">{`Select Accounts for ${getName(party)} requesting from ${getName(
+        serviceProvider || ''
+      )}`}</Modal.Header>
+      <Modal.Content>
+        {!custodyServices.length && (allocationAccountNeeded || accountNeeded) ? (
+          <>This party must have at least one Custody service</>
+        ) : (
+          !!serviceProvider && (
+            <div>
+              {_.toPairs(accountInfos).map(([k, accountInfo]) => {
+                const providerOptions =
+                  accountExists(accountInfo) || accountRequestExists(accountInfo)
+                    ? []
+                    : custodyServices.map(svc => {
+                        return {
+                          key: `REQUEST_${svc.payload.provider}_${accountInfo.accountLabel}`,
+                          label: `Request account from: ${getName(svc.payload.provider)}`,
+                          value: `${accountRequestPrepend}_${svc.payload.provider}`,
+                        };
+                      });
+                const accountOptions =
+                  accountInfo.accountType === AccountType.REGULAR
+                    ? accountNames
+                    : allocationAccountNames;
+                return (
+                  <>
+                    <Form.Select
+                      label={accountInfo.accountLabel}
+                      placeholder="Select..."
+                      required
+                      options={[...accountOptions, ...providerOptions]}
+                      value={accountNamesState[k]}
+                      onChange={(_, change) => {
+                        const [prepend, ...rest] = (change.value as string).split('_');
+                        if (prepend === accountRequestPrepend) {
+                          requestAccount(rest[0], accountInfo);
+                        } else {
+                          setAccountNamesState(prev => {
+                            let copy = { ...prev };
+                            copy[k] = change.value as string;
+                            return copy;
+                          });
+                        }
+                      }}
+                    />
+                    {accountRequestExists(accountInfo) && <p>Account request pending...</p>}
+                  </>
+                );
+              })}
+            </div>
+          )
+        )}
+      </Modal.Content>
+      <Modal.Actions>
+        <Button
+          className="ghost warning"
+          color="black"
+          onClick={() => {
+            onCancel();
+            setOpen(false);
+            setAccountNamesState(emptyNamesState);
```

Comment:
curious if this happens since the element leaves the DOM on line 243

Reply:
yeah, seems to work the same without it, makes sense.